### ISSUE TITLE
when switching keyboard language while Nvda focus is in an editbox, Nvda does not speek.

### ISSUE THREAD

**Issue by hamidrezaabroshan** [nvaccess/nvda#13630]
### Steps to reproduce:
Open notepad or any edit box and change keyboard language.
Note, I primarily using English and Persian.

### Actual behavior:
after switching keyboard lang, you have to move to another window to hear Nvda's voice
### Expected behavior:
after switching lang, you can continue typeing without any interruption.
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2021.3.5
#### Windows version:
 win 10Version 21H2 (OS Build 19044.1645)

#### Does the issue still occur after restarting your computer?
yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no.
#### If NVDA add-ons are disabled, is your problem still occurring?
yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes.
Nvda log:
NVDA initialized
ERROR - eventHandler.executeEvent (10:22:47.406) - MainThread (15084):
error executing event: typedCharacter on <NVDAObjects.UIA.ListItem object at 0x09941570> with extra args of {'ch': 'س'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 284, in executeEvent
  File "eventHandler.pyc", line 98, in __init__
  File "eventHandler.pyc", line 107, in next
  File "NVDAObjects\__init__.pyc", line 1118, in event_typedCharacter
  File "speech\speech.pyc", line 1045, in speakTypedCharacters
  File "api.pyc", line 334, in isTypingProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 960, in _get_isProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1545, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 927, in _getUIACacheablePropertyValue
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 30, in __call__
_ctypes.COMError: (-2147418094, 'The callee (server [not server application]) is not available and disappeared; all connections are invalid. The call did not execute.', (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (10:22:48.296) - MainThread (15084):
error executing event: typedCharacter on <NVDAObjects.UIA.ListItem object at 0x09941570> with extra args of {'ch': '\x08'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 284, in executeEvent
  File "eventHandler.pyc", line 98, in __init__
  File "eventHandler.pyc", line 107, in next
  File "NVDAObjects\__init__.pyc", line 1118, in event_typedCharacter
  File "speech\speech.pyc", line 1045, in speakTypedCharacters
  File "api.pyc", line 334, in isTypingProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 960, in _get_isProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1545, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 927, in _getUIACacheablePropertyValue
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 30, in __call__
_ctypes.COMError: (-2147418094, 'The callee (server [not server application]) is not available and disappeared; all connections are invalid. The call did not execute.', (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (10:22:50.460) - MainThread (15084):
error executing event: typedCharacter on <NVDAObjects.UIA.ListItem object at 0x09941570> with extra args of {'ch': 'j'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 284, in executeEvent
  File "eventHandler.pyc", line 98, in __init__
  File "eventHandler.pyc", line 107, in next
  File "NVDAObjects\__init__.pyc", line 1118, in event_typedCharacter
  File "speech\speech.pyc", line 1045, in speakTypedCharacters
  File "api.pyc", line 334, in isTypingProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 960, in _get_isProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1545, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 927, in _getUIACacheablePropertyValue
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 30, in __call__
_ctypes.COMError: (-2147418094, 'The callee (server [not server application]) is not available and disappeared; all connections are invalid. The call did not execute.', (None, None, None, 0, None))
ERROR - eventHandler.executeEvent (10:22:51.315) - MainThread (15084):
error executing event: typedCharacter on <NVDAObjects.UIA.ListItem object at 0x09941570> with extra args of {'ch': '\x08'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 284, in executeEvent
  File "eventHandler.pyc", line 98, in __init__
  File "eventHandler.pyc", line 107, in next
  File "NVDAObjects\__init__.pyc", line 1118, in event_typedCharacter
  File "speech\speech.pyc", line 1045, in speakTypedCharacters
  File "api.pyc", line 334, in isTypingProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 960, in _get_isProtected
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1545, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 927, in _getUIACacheablePropertyValue
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 30, in __call__
_ctypes.COMError: (-2147418094, 'The callee (server [not server application]) is not available and disappeared; all connections are invalid. The call did not execute.', (None, None, None, 0, None))
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (10:22:54.560) - MainThread (15084):
Developer info for navigator object:
name: 'NVDA'
role: Role.LISTITEM
roleText: None
states: State.FOCUSABLE, State.SELECTABLE, State.FOCUSED, State.SELECTED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_SysListView32EmittingDuplicateFocusEventsListItemIAccessible object at 0x09707E50>
Python class mro: (<class 'NVDAObjects.Dynamic_SysListView32EmittingDuplicateFocusEventsListItemIAccessible'>, <class 'appModules.explorer.SysListView32EmittingDuplicateFocusEvents'>, <class 'NVDAObjects.IAccessible.sysListView32.ListItem'>, <class 'NVDAObjects.behaviors.RowWithFakeNavigation'>, <class 'NVDAObjects.behaviors.RowWithoutCellObjects'>, <class 'NVDAObjects.IAccessible.sysListView32.ListItemWithoutColumnSupport'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=0, top=866, width=112, height=80)
value: None
appModule: <'explorer' (appName 'explorer', process ID 6424) at address 5f84550>
appModule.productName: 'Microsoft® Windows® Operating System'
appModule.productVersion: '10.0.19041.1620'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 66006
windowClassName: 'SysListView32'
windowControlID: 1
windowStyle: 1442855488
extendedWindowStyle: 0
windowThreadID: 16224
windowText: 'FolderView'
displayText: 'NVDA'
IAccessibleObject: <POINTER(IAccessible) ptr=0x5aaec30 at 98f13a0>
IAccessibleChildID: 8
IAccessible event parameters: windowHandle=66006, objectID=-4, childID=8
IAccessible accName: 'NVDA'
IAccessible accRole: ROLE_SYSTEM_LISTITEM
IAccessible accState: STATE_SYSTEM_SELECTED, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_MULTISELECTABLE, STATE_SYSTEM_VALID (19922950)
IAccessible accDescription: exception: (-2147352573, 'Member not found.', (None, None, None, 0, None))
IAccessible accValue: None

Thanks.

**Comment by CyrilleB79** [comment]
Confirmed on my side on NVDA 2022.1beta3.
It was already present in NVDA 2021.3.5 and maybe before.

Note that the issue does not always occur.
* If you first press alt to perform the alt+shift gesture, it seems that the issue does not occur if you quick enough to press shift, release it and release alt key.
* the issue seems to be reproducible if you press alt, press shift, release shift, wait at least one second and release alt.

When the issue occurs, I can see visually a list of the installed keyboard languages and the selection in this list being changed. When the issue is not reproduced, because I have executed the shortcut quickly enough, this list does not seem to appear. To be confirmed with better eyes than mine.

**Comment by Brian1Gaff** [comment]
Actually, I think this might also randomly occur when other displays overlay 
the edit field as well. Its very erratic and has been there for as long as I 
can remember. I now nearly always alt tab and come back its almost  second 
nature, but I cannot actually figure out what the controlling factor is. It 
does similar things in Windows 7, and wondered if it might actually be 
windows api issue.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by nikosdemetriou** [comment]
Hello.
I am having this issue when pressing alt+shift to change between Greek and English.

When I press alt+shift nvda says:
Control centre window, keyboard layout window, language switching list grouping, list English united kingdom 1 of 3. 

Very few times, I could navigate the list of languages with the up and down arrows, but most of the time the arrow keys are silent.
I asked sighted people and they said that nothing visual comes up on the screen.

I have got the latest windows 11 installed on an hp laptop, and I am having this issue with nvda 2021.3.5, and the upcoming nvda 2022.1 beta3.

It doesn't make any difference if the addons are disabled or not.

I had a look at the settings, and they keyboard layout is unpinned from the control centre and the language bar is disabled.

I don't know what is this invisible dialogue that is detected by nvda.

My log is attached.
The strange thing about it, is the fact that the alt+shift combination is not mentioned in the log even I used it many times.

Thanks

Nikos
[Nikos nvda.log](https://github.com/nvaccess/nvda/files/8601218/Nikos.nvda.log)
